planning/behavior_path_planner/src/utils/path_safety_checker/safety_check.cpp

```diff
@@ -364,27 +364,58 @@ std::vector<Polygon2d> getCollidedPolygons(
 
   return collided_polygons;
 }
-bool checkCollisionWithExtraStoppingMargin(
-  const PathWithLaneId & ego_path, const PredictedObjects & dynamic_objects,
-  const double base_to_front, const double base_to_rear, const double width,
-  const double maximum_deceleration, const double collision_check_margin,
-  const double max_extra_stopping_margin)
+
+std::vector<Polygon2d> generatePolygonsWithStoppingAndInertialMargin(
+  const PathWithLaneId & ego_path, const double base_to_front, const double base_to_rear,
+  const double width, const double maximum_deceleration, const double max_extra_stopping_margin)
 {
-  for (const auto & p : ego_path.points) {
+  std::vector<Polygon2d> polygons;
+
+  for (size_t i = 0; i < ego_path.points.size(); ++i) {
+    const auto p = ego_path.points.at(i);
+
+    const double curvature = [&]() {
+      if (i == 0 || i == ego_path.points.size() - 1) {
+        return 0.0;
+      }
+      return tier4_autoware_utils::calcCurvature(
+        ego_path.points.at(i - 1).point.pose.position,
+        ego_path.points.at(i + 0).point.pose.position,
+        ego_path.points.at(i + 1).point.pose.position);
+    }();
```

Comment:
nits:

it seems ok, but when `(i == 0 || i == ego_path.points.size() - 1)` may be better to use prev/next curvature like 

```
template <class T>
std::vector<double> calcCurvature(const T & points)
{
  std::vector<double> curvature_vec(points.size());

  for (size_t i = 1; i < points.size() - 1; ++i) {
    const auto p1 = tier4_autoware_utils::getPoint(points.at(i - 1));
    const auto p2 = tier4_autoware_utils::getPoint(points.at(i));
    const auto p3 = tier4_autoware_utils::getPoint(points.at(i + 1));
    curvature_vec.at(i) = (tier4_autoware_utils::calcCurvature(p1, p2, p3));
  }
  curvature_vec.at(0) = curvature_vec.at(1);
  curvature_vec.at(curvature_vec.size() - 1) = curvature_vec.at(curvature_vec.size() - 2);

  return curvature_vec;
}
```